### PR TITLE
8331858: [nmt] VM.native_memory statistics should work in summary mode

### PR DESCRIPTION
**Pull request by SoniaZaldana**
Hi all, 

This PR addresses [8331858](https://bugs.openjdk.org/browse/JDK-8331858) enabling the "statistics" sub-option for jcmd VM.native_memory. 

Since [8256844](https://bugs.openjdk.org/browse/JDK-8256844), preinit state gets printed under "statistics" and this is also relevant for the summary state. 

Testing:
- [x] Added test case passes. 

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8331858](https://bugs.openjdk.org/browse/JDK-8331858): [nmt] VM.native_memory statistics should work in summary mode (**Enhancement** - P4)

### Reviewers
 * [Thomas Stuefe](https://openjdk.org/census#stuefe) (@tstuefe - **Reviewer**)
 * [Johan Sjölen](https://openjdk.org/census#jsjolen) (@jdksjolen - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19179/head:pull/19179` \
`$ git checkout pull/19179`

Update a local copy of the PR: \
`$ git checkout pull/19179` \
`$ git pull https://git.openjdk.org/jdk.git pull/19179/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19179`

View PR using the GUI difftool: \
`$ git pr show -t 19179`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19179.diff">https://git.openjdk.org/jdk/pull/19179.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19179#issuecomment-2104815291)

### CODE REVIEW

File: src/hotspot/share/nmt/memTracker.cpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
- * Copyright (c) 2012, 2023, Oracle and/or its affiliates. All rights reserved.
- * Copyright (c) 2020, 2023 SAP SE. All rights reserved.
+ * Copyright (c) 2012, 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020, 2024 SAP SE. All rights reserved.
```
Comment (tstuefe) [inline src/hotspot/share/nmt/memTracker.cpp:3]: Please only adjust Oracle or RedHat copyrights, not from other companies

File: src/hotspot/share/nmt/nmtDCmd.cpp [hunk 2]
```diff
@@ -137,8 +137,11 @@ void NMTDCmd::execute(DCmdSource source, TRAPS) {
       output()->print_cr("No detail baseline for comparison");
     }
   } else if (_statistics.value()) {
-    if (check_detail_tracking_level(output())) {
+    NMT_TrackingLevel tracking = MemTracker::tracking_level();
+    if (tracking == NMT_detail || tracking == NMT_summary) {
```
Comment (tstuefe) [inline src/hotspot/share/nmt/nmtDCmd.cpp:141]: I think there is an "is_enabled" method in MemTracker that checks for both states. Please use that.

File: src/hotspot/share/nmt/nmtDCmd.cpp [hunk 3]
```diff
@@ -137,8 +137,11 @@ void NMTDCmd::execute(DCmdSource source, TRAPS) {
...
       MemTracker::tuning_statistics(output());
+    } else {
+      output()->print_cr("Detail or summary tracking is not enabled");
```
Comment (tstuefe) [inline src/hotspot/share/nmt/nmtDCmd.cpp:144]: I'd just say "NMT not enabled"

**Review by tstuefe — COMMENTED** [review]
Thanks for doing this. Minor nits, okay otherwise.

**Comment by SoniaZaldana** [comment]
Thanks @tstuefe! I made the updates.

**Review by tstuefe — APPROVED** [review]
Good for me.

File: src/hotspot/share/nmt/memTracker.cpp [hunk 4]
```diff
@@ -146,12 +146,17 @@ void MemTracker::report(bool summary_only, outputStream* output, size_t scale) {
...
+  if (_tracking_level == NMT_detail) {
+    out->print_cr("Malloc allocation site table size: %d",
+                  MallocSiteTable::hash_buckets());
+    out->print_cr("             Tracking stack depth: %d",
+                  NMT_TrackingStackDepth);
+    out->cr();
+    MallocSiteTable::print_tuning_statistics(out);
+    out->cr();
+  }
   out->print_cr("Preinit state:");
```
Comment (jdksjolen) [inline src/hotspot/share/nmt/memTracker.cpp:160]: Is there any functional difference here or is it only changing indentation?

Comment (SoniaZaldana) [reply src/hotspot/share/nmt/memTracker.cpp:160]: Hi @jdksjolen, are you referring to the indentation printing the tracking level? I can definitely revert the indentation there.

Comment (jdksjolen) [reply src/hotspot/share/nmt/memTracker.cpp:160]: Aha, I see, you added an `if` branch there! It was hard to see with the styling change on top. I think it's fine as is.

**Review by jdksjolen — APPROVED** [review]
Approved!

File: src/hotspot/share/nmt/memTracker.cpp (hunk 4 above)
Comment (SoniaZaldana) [reply src/hotspot/share/nmt/memTracker.cpp:160]: No worries, thanks for the review!

**Pull request closed without merging**